src/univariate/affine.jl

```diff
@@ -0,0 +1,241 @@
+log_abs_det = first ∘ logabsdet
+mult_identity(x::Real) = one(x)
+mult_identity(x::AbstractVector) = Diagonal(Ones(x))
+add_identity(x::Real) = zero(x)
+add_identity(x::AbstractArray) = Zeros(x)
+
+"""
+    AffineDistribution(μ, σ, base_dist)
+
+A shifted and scaled (affinely transformed) version of `base_dist`.
+
+If ``Z`` is a random variable with distribution `base_dist`, then `AffineDistribution` with
+parameters `μ`, `σ`, `base_dist` is the distribution of the random variable
+```math
+X = μ + σ * Z
+```
+
+If `base_dist` is a discrete distribution, the probability mass function of the transformed 
+distribution is given by
+```math
+P(X = x) = P\\left(Z = \\frac{x-μ}{σ} \\right).
+```
+
+If `base_dist` is a continuous distribution, the probability density function of
+the transformed distribution is given by
+```math
+f(x) = \\frac{1}{σ} base_dist \\! \\left( \\frac{x-μ}{σ} \\right).
+```
+
+If `base_dist` falls under a location-scale family, `affine` will attempt to return a 
+transformed distribution of type `base_dist` if an `affine` method has been implemented. 
+Otherwise, it will fall back on the `AffineDistribution` wrapper type.
+
+```julia
+d = σ * (base_dist + μ)     # Create location-scale transformed distribution
+params(d)                   # Get the parameters, i.e. (μ, scale, base_dist)
+location(d)                 # Get the location parameter
+scale(d)                    # Get the scale parameter
+```
+"""
+struct AffineDistribution{
+        F<:VariateForm,
+        S<:ValueSupport,
+        Tμ<:ArrayLike,
+        Tσ<:ArrayLike,
+        D<:Distribution{F,S}
+    } <: Distribution{F,S}
+    μ::Tμ
+    σ::Tσ
+    base_dist::D
+
+    function AffineDistribution(
+        μ::Tμ, σ::Tσ, base_dist::D; 
+        check_args=true
+        ) where {
+            F<:VariateForm,
+            S<:ValueSupport,
+            D<:Distribution{F, S},
+            Tμ<:ArrayLike,
+            Tσ<:ArrayLike
+        }
+        !check_args || @check_args(AffineDistribution, !iszero(σ))
+        return new{F,S,Tμ,Tσ,D}(μ, σ, base_dist)
+    end
+end
+
+
+# Constructors
+
+"""
+    +(shift, base_dist)
+
+Return a version of `base_dist` that has been translated by `shift`, where `shift` can be a
+real number or (for a multivariate distribution) a vector.
+"""
+function Base.:+(μ::ArrayLike, base_dist::Distribution)
+    return AffineDistribution(μ, mult_identity(μ), base_dist)
+end
+
+"""
+    *(scale_factor, base_dist)
+
+Return a version of `base_dist` that has been scaled by `scale`, where `scale` can be a real
+number or (for a multivariate distribution) matrix.
+"""
+function Base.:*(σ::ArrayLike, base_dist::Distribution)
+    return AffineDistribution(add_identity(σ), σ, base_dist)
+end
```

Comment:
Very confused by the second method defined here -- is this intended to work for a general `ArrayLikeVariate`, and not just multivariate distributions returning vectors? I can't make heads or tails of what's going on with the sizes.

Reply:
(Also, the ability to use `σ::Real, base_dist::Distribution` for both multivariate and univariate distributions is intentional, since scalar multiplication is defined for all array sizes.)

Reply:
I forgot to mention it here but yes, this only works for multi- and matrixvariate functions. While we might want to add scalar scales for arbitrary arrayvariate distributions we don't want to support arbitrary distributions and arbitrary arrays that are compatible with the distribution.